--- "./eclipse.platform.swt/bundles/org.eclipse.swt/Eclipse SWT Browser/gtk/org/eclipse/swt/browser/BrowserFactory.java.orig"	2016-04-29 12:15:18.524027606 +0100
+++ "./eclipse.platform.swt/bundles/org.eclipse.swt/Eclipse SWT Browser/gtk/org/eclipse/swt/browser/BrowserFactory.java"	2016-04-29 12:20:19.696980780 +0100
@@ -16,9 +16,6 @@
 
 WebBrowser createWebBrowser (int style) {
 	boolean webkitInstalled = WebKit.IsInstalled ();
-	if ((style & SWT.MOZILLA) != 0 || (!webkitInstalled && (style & SWT.WEBKIT) == 0)) {
-		return new Mozilla ();
-	}
 	if (!webkitInstalled) return null;
 
 	return new WebKit ();
